modules/basegl/src/datavisualizer/volumeraycastvisualizer.cpp

```diff
@@ -95,17 +96,17 @@ std::vector<Processor*> VolumeRaycastVisualizer::addVisualizerNetwork(Outport* o
     auto vbb = net->addProcessor(util::makeProcessor<VolumeBoundingBox>(GP{8, 3}));
     auto lrp = net->addProcessor(util::makeProcessor<LineRendererProcessor>(GP{8, 6}));
 
-    static_cast<FloatVec4Property*>(bak->getPropertyByIdentifier("bgColor1"))
+    dynamic_cast<FloatVec4Property*>(bak->getPropertyByIdentifier("bgColor1"))
         ->set(vec4(0.443f, 0.482f, 0.6f, 1.0f));
-    static_cast<FloatVec4Property*>(bak->getPropertyByIdentifier("bgColor2"))
+    dynamic_cast<FloatVec4Property*>(bak->getPropertyByIdentifier("bgColor2"))
         ->set(vec4(0.831f, 0.831f, 0.831f, 1.0f));
 
     // set shading mode in volume raycaster to 'no shading'
-    static_cast<OptionPropertyInt*>(vrc->getPropertyByIdentifier("shadingMode", true))->set(0);
+    dynamic_cast<OptionPropertyInt*>(vrc->getPropertyByIdentifier("shadingMode", true))->set(0);
 
-    static_cast<FloatProperty*>(lrp->getPropertyByPath({{"lineSettings"}, {"lineWidth"}}))
+    dynamic_cast<FloatProperty*>(lrp->getPropertyByPath({{"lineSettings"}, {"lineWidth"}}))
         ->set(1.5f);
-    static_cast<FloatVec3Property*>(vrc->getPropertyByIdentifier("lookFrom", true))
+    dynamic_cast<FloatVec3RefProperty*>(vrc->getPropertyByIdentifier("lookFrom", true))
```

Comment:
We should have a common interface for these if we don’t already

Reply:
what do you mean? an interface for what?

Reply:
I don't have access to the camera in the VolumeRaycaster it is a protected member...

Reply:
For OrdinalProperty/OrdinalRefProperty in this case it does not make sense to need to separate between the two

Reply:
They don't have identical interfaces, sometime I guess one could get away with using a common interface, but then you would still need to cast to that one, and I don't think we do that very often anyway.